api/src/main/java/org/motechproject/nms/api/web/MobileAcademyController.java

```diff
@@ -0,0 +1,131 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+
+/**
+ * Mobile Academy controller
+ */
+@RequestMapping("mobileacademy")
+@Controller
+public class MobileAcademyController extends BaseController {
+
+    /**
+     * MA service to handle all business logic
+     */
+    private MobileAcademyService mobileAcademyService;
+
+    /**
+     * Constructor for controller
+     * @param mobileAcademyService mobile academy service
+     */
+    @Autowired
+    public MobileAcademyController(MobileAcademyService mobileAcademyService) {
+        this.mobileAcademyService = mobileAcademyService;
+    }
+
+    /**
+     *
+     * 2.2.2.1 Get MA Course – Request
+     *
+     * Get course
+     * @return course response object
+     */
+    @RequestMapping(
+            value = "/course",
+            method = RequestMethod.GET)
+    @ResponseBody
+    public CourseResponse getCourse() {
+
+        // TODO: hook up to get course in MA service
+        return new CourseResponse();
+    }
+
+    /**
+     * Get the version of the course
+     * @return Integer representing the timestamp since epoch
+     */
+    @RequestMapping(
+            value = "/courseVersion",
+            method = RequestMethod.GET)
+    @ResponseBody
+    public CourseVersionResponse getCourseVersion() {
+        
+        return new CourseVersionResponse(mobileAcademyService.getCourseVersion());
+    }
+
+    /**
+     * Get bookmark for a user
+     * @param callingNumber number of the caller
+     * @param callId unique tracking id for the call
+     * @return serialized json bookmark response
+     */
+    @RequestMapping(
+            value = "/bookmarkWithScore",
+            method = RequestMethod.GET)
+    public GetBookmarkResponse getBookmarkWithScore(@RequestParam Long callingNumber,
+                                                 @RequestParam Long callId) {
+
+        return new GetBookmarkResponse();
+    }
+
+    /**
+     * Save a bookmark for a user
+     * @param bookmarkRequest info about the user for bookmark save
+     * @return OK or exception
+     */
+    @RequestMapping(
+            value = "/bookmarkWithScore",
+            method = RequestMethod.POST,
+            headers = { "Content-type=application/json" })
+    @ResponseStatus(HttpStatus.OK)
+    public void saveBookmarkWithScore(@RequestBody SaveBookmarkRequest bookmarkRequest) {
+
+        Long callingNumber = bookmarkRequest.getCallingNumber();
+        if (callingNumber == null || callingNumber < 1000000000) {
+            throw new IllegalArgumentException("Not a valid calling number: " + callingNumber);
+        }
+        if (bookmarkRequest.getCallId() == null) {
+            throw new IllegalArgumentException("Invalid callId " + bookmarkRequest.getCallId());
+        }
```

Comment:
See the validation methods in BaseController that you can reuse for these parameter validations.

Also note that the string passed to IllegalArgumentException should be of the format "<failureReason: description>".